apps/src/blockly/addons/cdoFieldBitmap.ts

```diff
@@ -11,28 +22,67 @@ export class CdoFieldBitmap extends FieldBitmap {
    * @param {object | null} config - Additional configuration options, can be an object or null/undefined.
    */
   constructor(
-    value?: number[][] | null,
-    options?: object | null,
-    config?: {fieldHeight?: number} | null
+    value: number[][] | typeof Blockly.Field.SKIP_SETUP,
+    options?: Blockly.FieldValidator<number[][]>,
+    config?: FieldBitmapFromJsonConfig
   ) {
     super(value, options, config);
-    this.fieldHeight = config?.fieldHeight;
   }
 
   /**
-   * Called when a new value has been validated and is about to be set.
-   * We extend this to ensure that the field height is always respected.
-   * See: https://github.com/google/blockly-samples/issues/2372
-   * @param newValue The value that's about to be set.
+   * Show the bitmap editor dialog.
+   *
+   * @param e Optional mouse event that triggered the field to open, or
+   *    undefined if triggered programmatically.
+   */
+  // eslint-disable-next-line
+  protected showEditor_(this: typeof FieldBitmap, e?: Event) {
+    super.showEditor_();
+
+    // Store row and column indices on each pixel button.
+    const pixelContainer = document.querySelector('.pixelContainer');
+    pixelContainer?.querySelectorAll('.pixelRow').forEach((row, rowIndex) => {
+      row.querySelectorAll('.pixelButton').forEach((button, colIndex) => {
+        // Set the custom data attributes for row and column indices
+        button.setAttribute('data-row', rowIndex.toString());
+        button.setAttribute('data-col', colIndex.toString());
+      });
+    });
+
+    // Handle dragging into a pixel when pointer is down. In the base class,
+    // mouseenter events are bound to the individual pixel buttons, but this
+    // isn't compatible with touch devices.
```

Comment:
Is it worth linking to the blockly ticket here for tracking/cleanup purposes?

Reply:
Yes, great call!